string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,111 @@
+//    - clear the current word
+//  - if the current word lowercase matches api
+//    - add API to the output and current character
+//    - clear the current word
+//  - if the second to last character in the output is a colon
+//    - capitalize the current word, add the word to the output, then current character
+//    - clear the current word
+//  - if the current word is four characters or more
+//    - capitalize the current word, then add the word and current character to the output
+//    - clear the current word
+//  - if the current word is less than four characters
+//    - look at each character in the current word
+//      - if the current word lowercase matches any word in the exceptions list
+//        - add the word, in lowercase, to the output, then current character of the title
+//        - clear the current word
+//    - if the current word wasn't added to the output
+//      - capitalize the current word, then add the word and current character to the output
+//      - clear the current word
+//  - otherwise, add the current character to the current word
+// if the current word lowercase matches javascript
+//  - add JavaScript to the output
+// if the current word lowercase matches api
+//  - add API to the output
+// otherwise, capitalize the current word and add it to the output
+// return the output
+
+function titleCase(title) {
+  var apaTitle = '';
+  var currentWord = '';
+  var exceptions = ['and', 'or', 'nor', 'but', 'a', 'an', 'the', 'as', 'at', 'by', 'for', 'in', 'of', 'on', 'per', 'to'];
+
+  function capitalize(word) {
+    var capitalizedWord = '';
+
+    capitalizedWord += word[0].toUpperCase();
+    for (var i = 1; i < word.length; i++) {
+      if (capitalizedWord[capitalizedWord.length - 1] === '-') {
+        capitalizedWord += word[i].toUpperCase();
+      } else {
+        capitalizedWord += word[i].toLowerCase();
+      }
+    }
+    return capitalizedWord;
+  }
+
+  for (var char = 0; char < title.length; char++) {
+    if (title[char] === ' ') {
+      if (currentWord.toLowerCase() === 'javascript') {
+        apaTitle += 'JavaScript' + title[char];
+        currentWord = '';
+      } else if (currentWord.toLowerCase() === 'api') {
+        apaTitle += 'API' + title[char];
+        currentWord = '';
+      } else if (currentWord.toLowerCase() === 'javascript:') {
+        apaTitle += 'JavaScript:' + title[char];
+        currentWord = '';
+      } else if (currentWord.toLowerCase() === 'api:') {
+        apaTitle += 'API:' + title[char];
+        currentWord = '';
+      } else if (apaTitle === '') {
+        apaTitle += capitalize(currentWord) + title[char];
+        currentWord = '';
+      } else if (apaTitle[apaTitle.length - 2] === ':') {
+        apaTitle += capitalize(currentWord) + title[char];
+        currentWord = '';
+      } else if (currentWord.length >= 4) {
+        apaTitle += capitalize(currentWord) + title[char];
+        currentWord = '';
```

Comment:
The work for each of these conditional blocks is the same, they could be merged. Would this work as an `else` block for this conditional? I think it could 🤔